src/main/java/org/candlepin/subscriptions/resource/InstancesResource.java

```diff
@@ -203,32 +204,27 @@ protected void validateBeginningAndEndingDates(OffsetDateTime beginning, OffsetD
   }
 
   private InstanceData asTallyHostViewApiInstance(
-      Host host, String monthId, List<String> measurements) {
+      TallyInstanceView tallyInstanceView, List<String> measurements) {
     var instance = new InstanceData();
     List<Double> measurementList = new ArrayList<>();
-    instance.setId(host.getInstanceId());
-    instance.setDisplayName(host.getDisplayName());
-    if (Objects.nonNull(host.getBillingProvider())) {
-      instance.setBillingProvider(host.getBillingProvider().asOpenApiEnum());
+    instance.setId(tallyInstanceView.getKey().getInstanceId().toString());
```

Comment:
I assume this won't throw a NPE. Correct?

Reply:
Since its part of they key it has to be non null so should be safe.